rfcs/text/0008-threat-intel.md

```diff
@@ -127,13 +222,19 @@ Stage 2: Identifies scope of impact of changes. Are breaking changes required? S
  * ECS project (e.g. docs, tooling)
 The goal here is to research and understand the impact of these changes on users in the community and development teams across Elastic. 2-5 sentences each.
 -->
+ * Ingestion mechanism: Primary ingestion mechanisms will be Filebeat modules and Ingest Packages. There will be no impact on ingestion mechanisms.
+ * Usage mechanism: The primary use of the proposed ECS fields and values is through Elastic Security solution. In 7.10 we released Indicator match rule to support the use of the proposed new fields and values.
 
 ## Concerns
 
 <!--
 Stage 1: Identify potential concerns, implementation challenges, or complexity. Spend some time on this. Play devil's advocate. Try to identify the sort of non-obvious challenges that tend to surface later. The goal here is to surface risks early, allow everyone the time to work through them, and ultimately document resolution for posterity's sake.
 -->
-There is a proposal to nest all IoC fields under `threat.ioc.*` instead of the current `threat.* structure.` This would make it consistent with taxonomy structure for `threat.tactic.*` and `threat.techinique.*` . This needs to be resovled in Stage 2 of the RFC process.
+
+1. Identified in Stage 1: There is a proposal to nest all IoC fields under `threat.ioc.*` instead of the current `threat.* structure.` This would make it consistent with taxonomy structure for `threat.tactic.*` and `threat.techinique.*` . This needs to be resovled in Stage 2 of the RFC process.
+ * Proposed resolution: Nest all IoC fields under `threat.ioc.*`
+2. How to use `event.module`
```

Comment:
The `event.module` is to represent where the data is coming from, to stay with this purpose and to have a "nice" way of displaying where threat data is coming from it could be beneficial to leave this open with the guidance of examples.

E.g.:
For abuse.ch event would be populated as follows:
```json
"event": {
  "module": "abuse",
  "dataset": "abuse.urlhaus"
}
```

Reply:
I think the concern around `event.module` is that there won't be a _module_ (in the [traditional sense](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-modules.html)) for abuse.ch (or OTX, or Limo, etc.). A thought was to use `event.provider` to capture the source of the information and then use `event.dataset` in a similar way as you've laid out.

What do you think?

Reply:
Combining this with above the other discussions; How about using a `threat.ioc.provider`?

Reply:
When considering the enrichment document, I'm seeing how nesting these under `threat.ioc.*` makes more and more sense. 

In the enriched document, `event.provider` would be the name of the software that generated the event (e.g. Sysmon, httpd), or of a subsystem of the operating system (kernel, Microsoft-Windows-Security-Auditing) with the `threat.ioc.provider` being the source of the threat information.

Reply:
Given the discussion about enrichment use case and the arguments made for moving `.module` , `.dataset` and `.provider` under `threat.ioc.*` I will propose that we reflect this change in an updated PR.

Reply:
This looks to be completed.

Reply:
@peasead needs to add `threat.ioc.{module,dataset,provider}` to the `threat.yml` file.

@dcode will update the [example document](https://github.com/elastic/ecs/blob/1347f2ba4c0ef6c00d5ffccee7fa2ad854f631d0/rfcs/text/0008-threat-intel.md#usage)